### ISSUE TITLE
Add "volume" input/output resource

### ISSUE THREAD

**Issue by abayer** [tektoncd/pipeline#1062]
Related to but not exactly the same as #924. =)

In Jenkins X, we're going to be running two `PipelineRun`s of different `Pipeline`s for each build - the first will dynamically generate the second `Pipeline` from the `jenkins-x.yml` and build packs etc in the repo and potentially make changes to other files in the workspace, while the second will run the actual build. To do this, we need to be able to pass the modified files from the first `PipelineRun`'s workspace to the second `PipelineRun` (in the case of pull requests, we also merge the pull request into master locally in both `PipelineRun`s currently, and it would be nice to avoid doing that in the second one). We considered using a `gcs` resource for this, but that would break Jenkins X running on anything but GCP, which isn't an option, and would require bucket configuration, etc.

So we've got https://github.com/jenkins-x/jx/issues/4660 - we'd like to take the workspace at the end of the first `PipelineRun` and use that as the input to the second one. I thought about trying to just reuse the first `PipelineRun`'s artifact storage PVC, but decided it was better to make a new PVC and use that, so that we don't have to mess with the cleanup logic for artifact storage PVCs. Anyway, I've got some experimental work on this and will hopefully have a proof-of-concept PR up this week.

**Comment by bobcatfish** [comment]
After implementing this (har har) we realized that we might not need it after all, which I started to discuss in https://github.com/tektoncd/pipeline/pull/1417#issuecomment-546436595 

We were creating the VolumeResource to solve several problems that it looks like will be solved other ways:
- @skaegi identified a lot of pain points around Volumes and VolumeMounts in Tasks which should be fixed in #1438
- @dlorenc and I found a bunch of issues with the way PVC linking worked (via the bug we tried to fix in #937 leading to re-thinking PVCs in #1076 
- In general we've started re-thinking PipelineResources via #1272 
- @abayer requested in this issue (#1062) a way to use the "workspace" (data on PVC) from one PipelineRun in another PipelineRun (jenkins-x/jx#4660)

@sbwsg and @dlorenc have a new design to propose for #1272 and #1076 which will make it so that:
- PipelineResources can declare whether they need to be backed by underlying storage such as a PVC or bucket
- Tekton admins can configure what underlying storage mechanism is used; if none is configured, PVCs will be automatically created and destroyed for this
- PipelineRuns can override the storage mechanism and provide their own PVC if desired <-- this should handle the use case described by @abayer , tho it would mean the user (or jenkins x in this case) would need to create the PVC before the first PipelineRun

So long story short, between the new design (above is a teaser, watch #1272 and #1076 for more details!) and @skaegi 's proposal in #1438 we think we can meet all the use cases we know of _without_ a Volume Resource but please add comments if you disagree and we can re-open!

(+@ravikiranbukka)
